crates/primitives/src/transaction/tx_type.rs

```diff
@@ -73,6 +73,11 @@ impl TxType {
             Self::Deposit => false,
         }
     }
+
+    /// Convert from alloy consensus [`alloy_consensus::TxType`] to [`TxType`].
+    pub fn from_alloy_tx_type(value: alloy_consensus::TxType) -> Self {
+        value.into()
+    }
```

Comment:
@mattsse Here I also think that we should avoid to duplicate the definition of `TxType` which is already in alloy and probably does not need to be redefined in reth. The only blocking point will be the deposit variant of Optimism which cannot be handled by alloy which is only focused on ethereum... We probably need to think about it.